language/stdlib/modules/Libra.move

```diff
@@ -380,10 +380,10 @@ module Libra {
         preburn_address: address,
         _capability: &BurnCapability<CoinType>
     ) acquires CurrencyInfo {
+        let currency_code = currency_code<CoinType>();
         // destroy the coin at the head of the preburn queue
         let Libra { value } = Vector::remove(&mut preburn.requests, 0);
         // update the market cap
-        let currency_code = currency_code<CoinType>();
```

Comment:
I changed the order of code so that the unpack of the coin and the update of `total_value` are done together. This way the invariant that `sum_of_coin_values == total_value` will hold for `currency_code<CoinType>`.